addons/hr/models/res_partner.py

```diff
@@ -62,3 +64,22 @@ def _compute_employee(self):
         employees = {employee for [employee] in employee_data}
         for partner in self:
             partner.employee = partner in employees
+
+    def _to_store_defaults(self):
+
+        def get_employee_value(field_to_get):
+            def get_for_partner(partner):
+                # sudo: res.partner-fetch employees from the partner
+                matching_employees = partner.sudo().employee_ids.filtered(
+                    lambda employee: employee.company_id == partner.company_id
+                )
+                values = matching_employees.mapped(field_to_get) or [False]
+                return values[0]
+            return get_for_partner
+
+        fields = {
+            "department": get_employee_value("department_id.name"),
+            "designation": get_employee_value("job_title"),
+            "workplace": get_employee_value("work_location_name"),
+        }
+        return super()._to_store_defaults() + list(starmap(Store.Attr, fields.items()))
```

Comment:
This should not be added to store in all flows, only the one that needs them.

Reply:
Also we don't want to store employee value on partner, but introduce the employee model and make a relation between the two.

Reply:
We might want to make a first PR without the HR overrides it is allows moving faster and refining this part later.